### PR TITLE
feat(sdk-coin-xlm): root key support for XLM recovery and refactor XLM utils

### PR DESCRIPTION
**Pull request by dwei-bitgo**
TICKET: [WP-2094](https://bitgoinc.atlassian.net/browse/WP-2094)

[WP-2094]: https://bitgoinc.atlassian.net/browse/WP-2094?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/sdk-coin-xlm/test/unit/xlm.ts [hunk 1]
```diff
@@ -314,12 +314,12 @@ describe('XLM:', function () {
...
       pub: '37c6be1c677873e42a90f3d0ed3b0e4f3ba8591f285769d8c922f7024ae0c009',
-      prv: 'c034ca796a145e79acfac9bc9d97e4d3aaa42bfbe648b46bb1f3da794d7765fb',
+      prv: 'c034ca796a145e79acfac9bc9d97e4d3aaa42bfbe648b46bb1f3da794d7765fb37c6be1c677873e42a90f3d0ed3b0e4f3ba8591f285769d8c922f7024ae0c009',
     };
     // This key pair is the decoded version of the backupKeychain above
     const backupRootKeychain = {
       pub: 'b63de5ad57cc75b256483bc1c1de82689497eaf27a930896e7733f418acf9eca',
-      prv: '35a98c286af2c36283adcd47e92cc221b1cd24eeded852a21abfbbc65e98eb8a',
+      prv: '35a98c286af2c36283adcd47e92cc221b1cd24eeded852a21abfbbc65e98eb8ab63de5ad57cc75b256483bc1c1de82689497eaf27a930896e7733f418acf9eca',
     };
```
Comment (zahin-mohammad) [inline modules/sdk-coin-xlm/test/unit/xlm.ts:323]: Instead of changing these, would it make sense to create a new keypair in the new format? That way we still have tests for the old format.

File: modules/sdk-coin-xlm/src/lib/utils.ts [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
+ * Private key can be either a raw root private key (hex string) or a Stellar private key (prefixed with 'S').
+ * @param secret Either hex root private key or Stellar private key
+ * @returns Stellar keypair from the provided prv
+ */
+export function createStellarKeypairFromPrv(prv: string): stellar.Keypair {
+  if (prv.startsWith('S')) {
+    return stellar.Keypair.fromSecret(prv);
+  }
+  return stellar.Keypair.fromSecret(encodePrivateKey(Buffer.from(prv.slice(0, 64), 'hex')));
+}
```
Comment (zahin-mohammad) [inline modules/sdk-coin-xlm/src/lib/utils.ts:28]: Lets add tests for these utils.

File: modules/sdk-coin-xlm/src/lib/utils.ts [hunk 3]
```diff
@@ -1,22 +1,48 @@
...
+
 /**
  * Validates a Stellar public key
  * Stellar public keys are prefixed with 'G'
  * @param pub A Stellar public key to validate
  * @returns Whether the input is a valid Stellar public key
  */
-export function isValidStellarPub(pub: string): boolean {
+export function isValidStellarPublicKey(pub: string): boolean {
```
Comment (zahin-mohammad) [inline modules/sdk-coin-xlm/src/lib/utils.ts:36]: This is a breaking change. Keep the existing function, and mark it as `@deprecated` in the jsdocs, and create a new function for what you need. In the jsdocs mention the new function clients should use. 

This applies to all the other function name changes.

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-xlm/src/lib/utils.ts (hunk 2 above)
Comment (dwei-bitgo) [reply modules/sdk-coin-xlm/src/lib/utils.ts:28]: There are some tests for these, they are in the `utils.ts` test file at the bottom

File: modules/sdk-coin-xlm/test/unit/xlm.ts (hunk 1 above)
Comment (dwei-bitgo) [reply modules/sdk-coin-xlm/test/unit/xlm.ts:323]: Hmm, I changed these as I realized that root prvs are represented in `prv + pub` format, so I think that the original format of these test keychains actually would not occur

File: modules/sdk-coin-xlm/src/lib/utils.ts (hunk 3 above)
Comment (dwei-bitgo) [reply modules/sdk-coin-xlm/src/lib/utils.ts:48]: Updated that

File: modules/sdk-coin-xlm/test/unit/xlm.ts (hunk 1 above)
Comment (zahin-mohammad) [reply modules/sdk-coin-xlm/test/unit/xlm.ts:323]: It could be that we used to support this format but then changed it later, which would mean some old wallets would have that old format. I am not actually sure, but if the tests passed with the old format, I'd expect that it should still work? 

Are the old formats still valid for the tests they are used for or do they error out?

File: modules/sdk-coin-xlm/test/unit/utils.ts [hunk 4]
```diff
@@ -91,8 +91,46 @@ describe('Stellar Utils', () => {
...
       const decodedBackupPrv = Utils.decodePrivateKey(backupKeychain.prv).toString('hex');
 
-      decodedPrv.should.equal(rootKeychain.prv);
-      decodedBackupPrv.should.equal(backupRootKeychain.prv);
+      decodedPrv.should.equal(rootKeychain.prv.slice(0, 64));
+      decodedBackupPrv.should.equal(backupRootKeychain.prv.slice(0, 64));
+    });
+  });
+
+  describe('create Stellar keypairs', () => {
```
Comment (zahin-mohammad) [inline modules/sdk-coin-xlm/test/unit/utils.ts:99]: Can you add non-happy path tests as well? What errors do we expect for invalid input, what if input is empty, etc.

**Review by zahin-mohammad — DISMISSED** [review]
lgtm. You can add the extra tests either in this PR or another.

File: modules/sdk-coin-xlm/test/unit/xlm.ts (hunk 1 above)
Comment (dwei-bitgo) [reply modules/sdk-coin-xlm/test/unit/xlm.ts:323]: So these tests and this test data I added in the initial root key PR for XLM here: https://github.com/BitGo/BitGoJS/pull/4638

Before this, no XLM wallet was able to be created using root keys. I reason I updated these keys was because I realized that root keys generated in the OVC, specifically the private key part, are given in an encrypted version of the `prv + pub` 64-byte format, rather than that old 32-byte `prv` only one.

The old format does still pass the tests, since in the code the place they are used is calling a `slice(0, 64)` to extract the first 32-bytes of the hex string anyway.

File: modules/sdk-coin-xlm/test/unit/utils.ts (hunk 4 above)
Comment (dwei-bitgo) [reply modules/sdk-coin-xlm/test/unit/utils.ts:100]: Added these tests into this PR. Also refactored these methods to have better error handling

**Review by margueriteblair — APPROVED** [review]
